### PR TITLE
Feature 8

### PR DESCRIPTION
**Pull request by Namrata-J**
#9 here i have added all the wishlist and cart functionality

now, you can :
:- brief of Add To Cart button and Buy Now Button
-add the item to cart ( when clicked on add to cart button it adds the item to cart )
-when the item is added the add to cart gets toggled to added and when anyhow the item is removed from cart then it gets toggled again back to it's original text i.e. add to cart text
-clicking on add to cart multiple times didn't add multiple templates in cart section (i.e. the cart template remains the same and also the count)
-clicking on buy now button takes you to the cart or you can go their by clicking the icon up there in header

also,
:- brief about add to wishlist secondary icon
- when clicked on the icon, it adds it to the wishlist, and viceversa(i.e. when clicked again, removes it from the wishlist)
- it's color gets toggled accordingly(i.e red when item added to wishlist, and black when item not added to wishlist)

also,
:- brief about the wishlist section
- have added conditional card templates same as was done for the producctlisting page(i.e. conditional templates f0r trending, new and stockout products)
- now when clicked on move to cart button it adds it to cart and removes it from wishlist.
- if the item is already added to cart, then it only increases it's count and do not add a different template.
- when clicked on remove button it removes it from the wishlist.
- also in this whole process it does take care of the state of wishlist icon on product in productlisting page. and you can either remove it from the wishlist by clicking on remove button on wishlisted product or on secondary wishlist icon on product on product page
- when wishlist is empty it shows you different template with an image and link to productlisting page

also,
:- brief about cart section
- clicking on the move to wishlist button on item added to cart, moves it to wishlist and removes it from cart.
- quantity could be increased or decreased using increment and decrement buttons.
- quantity can't be decreased further 1
- when clicked on remove it removes it from cart
- pricedetails are showm accordingly with the items added in cart

also,
- have added counter functionality to icons on buttons. (wherein the count of products added are shown in badges on icons)

please review it, thankyou!!

live preview: https://aprize-ecom-site.netlify.app/

### CODE REVIEW

File: src/components/Products-In-Cart/productsInCart.css [hunk 1]
```diff
@@ -155,6 +155,18 @@
...
 }
 
+.ap_empty-cart-img-container {
+    display: flex;
+    flex-direction: column;
+    gap: 2rem;
+}
+
+.ap_empty-cart-img {
+    clip-path: circle();
```
Comment (shaksham08) [inline src/components/Products-In-Cart/productsInCart.css:165]: nice use of clip path

**Review by shaksham08 — APPROVED** [review]
code looks well structured

File: src/utils/productutilFuncs.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            }
+        } else {
+            if (product.isInStock) {
+                return "card-w-badge-type1 card-w-badge b-rad1"
+            } else {
+                return "card-w-badge-type3 card-w-badge b-rad1"
+            }
+        }
+    }
+}
```
Comment (JantuDeb) [inline src/utils/productutilFuncs.js]: This is good but you can try to remove some if statement. like in line 2 you have a check for `isNewArival` and you have the same check in line 3 also.

**Review by JantuDeb — APPROVED** [review]

File: src/utils/productutilFuncs.js (hunk 2 above)
Comment (Namrata-J) [reply src/utils/productutilFuncs.js]: yes, thanks for pointing it out!! will refactor the code here!!

**Pull request merged**